homeassistant/components/waterkotte_heatpump/entity.py

```diff
@@ -0,0 +1,38 @@
+"""base waterkotte entity."""
+from pywaterkotte.ecotouch import TagData
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import DeviceInfo, EntityDescription
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import EcotouchCoordinator
+from .const import DOMAIN
+
+
+class EcotouchEntity(CoordinatorEntity[EcotouchCoordinator]):
+    """Base representation of waterkotte entities."""
+
+    def __init__(
+        self,
+        coordinator: EcotouchCoordinator,
+        tag: TagData,
+        config_entry: ConfigEntry,
+        entity_config: EntityDescription,
+        device_info: DeviceInfo,
+    ) -> None:
+        """Initialize entity."""
+        super().__init__(coordinator)
+        self.entity_description = entity_config
+
+        self.config_entry = config_entry
+        self._tag = tag
+        self.entity_id = f"{DOMAIN}.{entity_config.key}"
+        self._attr_unique_id = f"{config_entry.unique_id}.{entity_config.key}"
+        coordinator.alltags.add(tag)
+        self._attr_device_info = device_info
+
```

Comment:
I think all new integrations are required to set _attr_has_entity_name and follow its naming rules. See: https://developers.home-assistant.io/blog/2022/07/10/entity_naming/

Reply:
should be fixed with commit [aea0760](https://github.com/home-assistant/core/pull/88140/commits/aea07609569f459b954d1d5cd22826a1f7540f93)